### ISSUE TITLE
Trying to use this in VITS

### ISSUE THREAD

**Issue by codeghees** [supertone-inc/super-monotonic-align#2]
Based on this [repo](https://github.com/jaywalnut310/vits/blob/2e561ba58618d021b5b8323d3765880f7e0ecfdb/models.py#L480). 
```
attn = (
                maximum_path(neg_cent, attn_mask.squeeze(1))
                .unsqueeze(1)
                .detach()
```

I imported SMAS:
```from super_monotonic_align import maximum_path```

Is this implementation correct? Normal MAS is working fine but this repo errors out. The alignment is not a diagonal. Should I be fixing something else?

**Comment by junjun3518** [comment]
Hi!
Can you show your result more?

**Comment by junjun3518** [comment]
Or can you check your attn_mask is contiguous?

**Comment by codeghees** [comment]
```
print("neg_cent", neg_cent.shape)
print("attn_mask", attn_mask.shape)
print("attn_mask contiguous", attn_mask.is_contiguous())
neg_cent torch.Size([24, 300, 73])
attn_mask torch.Size([24, 1, 300, 73])
attn_mask contiguous True
```
![image](https://github.com/user-attachments/assets/93efe7b0-2c38-43b5-bfbb-55e71f21f6f2)

**Comment by codeghees** [comment]
Running test.py
So far: 
```
> python test.py
Passed on shape=(32,16,16)
Passed on shape=(32,128,512)
Passed on shape=(32,256,1024)
Passed on shape=(32,511,2048)
```

It is still running though.

**Comment by junjun3518** [comment]
300 is text length and 73 is audio length?
It only consider text length<= audio length case

**Comment by codeghees** [comment]
opposite! 
```
x torch.Size([24, 192, 120])
m_p torch.Size([24, 192, 120])
z torch.Size([24, 192, 392])
z_p torch.Size([24, 192, 392])
neg_cent torch.Size([24, 392, 120])
attn_mask torch.Size([24, 1, 392, 120])
```

**Comment by codeghees** [comment]
I guess I need to re-order the tensors?

**Comment by junjun3518** [comment]
For now, I think it is a reason. 
I don't know why the order is changed, it seems like the dim order in VITS is changed from Glow-TTS.

```
# vits
with torch.no_grad():
      # negative cross-entropy
      s_p_sq_r = torch.exp(-2 * logs_p) # [b, d, t]
      neg_cent1 = torch.sum(-0.5 * math.log(2 * math.pi) - logs_p, [1], keepdim=True) # [b, 1, t_s]
      neg_cent2 = torch.matmul(-0.5 * (z_p ** 2).transpose(1, 2), s_p_sq_r) # [b, t_t, d] x [b, d, t_s] = [b, t_t, t_s]
      neg_cent3 = torch.matmul(z_p.transpose(1, 2), (m_p * s_p_sq_r)) # [b, t_t, d] x [b, d, t_s] = [b, t_t, t_s]
      neg_cent4 = torch.sum(-0.5 * (m_p ** 2) * s_p_sq_r, [1], keepdim=True) # [b, 1, t_s] 
      neg_cent = neg_cent1 + neg_cent2 + neg_cent3 + neg_cent4

#glow tts
with torch.no_grad():
        x_s_sq_r = torch.exp(-2 * x_logs)
        logp1 = torch.sum(-0.5 * math.log(2 * math.pi) - x_logs, [1]).unsqueeze(-1) # [b, t, 1]
        logp2 = torch.matmul(x_s_sq_r.transpose(1,2), -0.5 * (z ** 2)) # [b, t, d] x [b, d, t'] = [b, t, t']
        logp3 = torch.matmul((x_m * x_s_sq_r).transpose(1,2), z) # [b, t, d] x [b, d, t'] = [b, t, t']
        logp4 = torch.sum(-0.5 * (x_m ** 2) * x_s_sq_r, [1]).unsqueeze(-1) # [b, t, 1]
        logp = logp1 + logp2 + logp3 + logp4 # [b, t, t']
```

Please use it with transpose for a while.
Thank you for noticing, and we will gonna add [BST] version.

**Comment by codeghees** [comment]
it worked! thanks!

**Comment by junjun3518** [comment]
Thank you for checking @codeghees!
Since our goal is to make MAS efficient, we will release BST version soon and let you know.
